lib/graphql/schema/scalar.rb

```diff
@@ -49,6 +49,7 @@ def validate_non_null_input(value, ctx)
             err
           end
 
+          require 'graphql/coercion_error'
```

Comment:
I'm guessing this is because, previously, this code was _not_ executed while `graphql`'s own files were being loaded by Ruby. But after this change, some default values (perhaps from introspection-related code) are now taking this codepath at load-type. So, the load-time dependency situation changed. 

I encounter this from time to time when doing big refactors, too, and I consider it a code smell pointing at a larger bit of tech debt. When I started adding `require`s for this gem, I didn't really know what I was doing, I basically put them _all_ in `graphql.rb`. (Take a look, you'll see a bazillion `require`s in a row, and 🙀  they're order-dependent.) 

A better system would be if each file, y'know, contained the `require`s it needed in order to operate. It's possible that it would end up _not_ working, if there's some kind of cyclical dependency, but I'm not sure. I just haven't been able to _prioritize_ that work, since everything works fine, and _one_ more hack can't possibly make it any worse 😅 .